great_expectations/validation_operators/validation_operators.py

```diff
@@ -304,41 +305,55 @@ def run(
         elif not isinstance(run_id, RunIdentifier):
             run_id = RunIdentifier(run_name=run_name, run_time=run_time)
 
-        run_results = {}
-
-        for item in assets_to_validate:
-            run_result_obj = {}
-            batch = self._build_batch_from_item(item)
-
-            if hasattr(batch, "active_batch_id"):
-                batch_identifier = batch.active_batch_id
-            else:
-                batch_identifier = batch.batch_id
+        with AsyncExecutor(
```

Comment:
This is where multithreading occurs using the V2 API.

For the V3 API, multiple threads are used in `Checkpoint.run`.  With the V3 API, `len(assets_to_validate)` is always 1 here and providing that as `max_workers` ensures that no new threads are spawned here with the V3 API (since we already have multiple threads in `Checkpoint.run`).

Reply:
Added comment block in the code explaining this.